### PR TITLE
added api base tests, purchasing channel and bulk upload methods and …

### PR DESCRIPTION
**Pull request by ryan-symphony**
…tests

### CODE REVIEW

File: dist/classes/APIBase.js [hunk 1]
```diff
@@ -85,14 +85,15 @@ var APIBase = (function () {
...
+            form: params.form,
+            formData: params.formData
         };
         return this.doRequest(options);
     };
     APIBase.prototype.doRequest = function (options) {
         this.logger.info(("[ " + options.method + " ] : ") + options.uri);
         // request.debug = true;
-        options.headers = this.headers;
+        options.headers = _.assign({}, this.headers);
```
Comment (ryan-symphony) [inline dist/classes/APIBase.js]: Don't want to mutate the base header in case special request headers needed [not at the moment but in the future could cause weird side effects]

File: dist/classes/APIBase.js [hunk 2]
```diff
@@ -118,7 +119,7 @@ var APIBase = (function () {
     APIBase.prototype.doRequest = function (options) {
         this.logger.info(("[ " + options.method + " ] : ") + options.uri);
         //request.debug = true;
```
Comment (rileylnapier) [inline dist/classes/APIBase.js]: scope creep alert:  adding `request.debug = this.apiOptions.debug` would be cool

File: test/apibase.spec.js [hunk 3]
```diff
@@ -1,45 +1,142 @@
...
+
+  const baseRequestOptions = {
+    baseUrl: `https://manage.symphonycommerce.com/org/${baseParams.site}`,
+    callback: undefined,
+    uri: baseOptions.apiPath,
+    qs: '',
+    json: true,
+    headers: {
+        Accept: "application/json",
+        Authorization: "Basic dGVzdEB0ZXN0LmNvbTpwdw==",
```
Comment (rileylnapier) [inline test/apibase.spec.js]: env variable?

Comment (ryan-symphony) [reply test/apibase.spec.js]: Oh that's just a hash for the mock email I passed in? I got that by console logging the options generated for the mock request
